docker/docker-Ubuntu/Dockerfile

```diff
@@ -15,13 +15,13 @@
 # limitations under the License.
 #
 
-FROM ubuntu:21.10
+FROM ubuntu:22.04
 LABEL maintainer="Yanhui Zhao (yanhui.zhao@outlook.com)"
 
 RUN apt-get -y update && apt-get install -y software-properties-common && add-apt-repository ppa:openjdk-r/ppa && apt-get -y update && \
     apt-get install -y openjdk-17-jdk cmake check git pkg-config autoconf man build-essential gcc g++ uuid-dev pandoc devscripts flex doxygen maven
 
-RUN apt-get install -y libndctl-dev libpmem-dev libvmem-dev libpmemobj-dev libmemkind-dev
+RUN apt-get install -y libndctl-dev libpmem-dev libpmemobj-dev libmemkind-dev
```

Comment:
I'm not sure that it can get build if we remove this dependent library as the libmemkind-dev still uncompleted. thanks.

Reply:
libvmem-dev is not available on 22.04, so it temporary work around is to build from source. Please see other changes in this PR. Thanks!